agent/README.md

```diff
@@ -18,14 +18,15 @@ func main() {
 
 ## Agent options
 
-There are various options available to customize agent.
+There are various options available to customize agent:
 
 - `Version(ver, date, id)` sets version of the program
-- `QuitOnClose(chan)` sets signal used to quit running agent when closed
-- `QuitSignals(signals)` sets signals used to quit running agent (default: SIGINT, SIGTERM)
+- `QuitOnClose(chan)` sets signal used to quit the running agent when closed
+- `QuitSignals(signals)` sets signals used to quit the running agent (default: SIGINT, SIGTERM)
 - `StartTimeout(dur)/StopTimeout(dur)` sets start/stop timeout (defaults: 15s/5s)
-- add plugins to list of plugins managed by agent using:
-  - `Plugins(...)` adds just single plugins
-  - `AllPlugins(...)` adds plugin along with all of its plugin deps
+
+Add plugins to list of plugins managed by agent using:
+- `Plugins(...)` adds just single plugins
+- `AllPlugins(...)` adds plugin along with all of its plugin deps
```

Comment:
Since the plugin lookup seems to get more complicated now. It might be good idea to document all the possible situations which are supported by the lookup.

Currently the lookup will search for plugin dependencies in:
- all non-nil fields of a plugin that implement `infra.Plugin` interface
- including fields from all anonymous fields recursively
- including elements of from slice fields

Reply:
Agree, and want to prepare the new documentation/guide for the plugin lookup where this will be mentioned it more detail. I plan it as a new PR, so let's omit it here for now if you don't mind.